barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.hpp

```diff
@@ -70,6 +70,62 @@ class ClientIVC {
 
         size_t size() const { return mega_proof.size() + goblin_proof.size(); }
 
+        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1299): The following msgpack methods are generic
+        // and should leverage some kind of shared msgpack utility.
+        msgpack::sbuffer to_msgpack_buffer() const
+        {
+            msgpack::sbuffer buffer;
+            msgpack::pack(buffer, *this);
+            return buffer;
+        }
```

Comment:
Suspect that this didn't add any new headers (but also in general this should be in a cpp file as msgpack stuff that isn't our thin header include should avoid header pollution)

Reply:
yeah I definitely don't want this in here. When you're back we should discuss how to standardize things, assuming we move to more msgpack use in general. I saw you had some established patterns using `msgpack::packer` - maybe this can just fit into that but wasn't sure it was the right thing